shared/KOSocket.cpp

```diff
@@ -139,23 +141,27 @@ bool KOSocket::DecryptPacket(uint8 *in_stream, Packet & pkt)
 
 bool KOSocket::Send(Packet * pkt) 
 {
-	if (!IsConnected() || pkt->size() + 1 > GetWriteBuffer().GetAllocatedSize())
+	if (!IsConnected()
+		|| pkt->size() + 7 > MAX_PACKET_SEND_SIZE)
 		return false;
 
-	bool r;
+	WSABUF buf;
+	DWORD w_length = 0;
+	DWORD flags = 0;
+	BYTE out_stream[MAX_PACKET_SEND_SIZE];
+	BYTE Final_Packet[MAX_PACKET_SEND_SIZE];
+	uint32 Final_Size = 0;
+	memset(out_stream, 0x00, MAX_PACKET_SEND_SIZE);
+	memset(Final_Packet, 0x00, MAX_PACKET_SEND_SIZE);
 
-	uint8 opcode = pkt->GetOpcode();
-	uint8 * out_stream = nullptr;
 	uint16 len = (uint16)(pkt->size() + 1);
 
 	if (isCryptoEnabled())
 	{
 		len += 5;
 
-		out_stream = new uint8[len];
```

Comment:
Can you explain what was the decision from altering the heap-allocated packet buffer to stack-allocated?

I feel like this is wrong and a waste of resources. I'd personally use an `std::vector<uint8_t>` of bytes. If you can elaborate more on this and explain why this is healthy for the socket system, maybe we can consider this change with a couple of alterations if they make any sense.